.github/workflows/trigger-release-candidate-build-and-deploy.yml

```diff
@@ -28,9 +28,30 @@ jobs:
 
           echo "Latest commit on 'staging' branch is: $(git log -1 --format='%H' staging)"
 
-          echo "Latest commit on 'main' branch is: $(git log -1 --format='%H' main)"
+          # echo "Latest commit on 'main' branch is: $(git log -1 --format='%H' main)"
+          echo "Latest commit on 'psridharan/trigger-stage-deploy' branch is: $(git log -1 --format='%H' psridharan/trigger-stage-deploy)"
 
-          echo "About to merge 'main' branch into 'staging' branch"
-          git merge --verbose main -m "Merging main branch into staging branch"
+          # echo "About to merge 'main' branch into 'staging' branch"
+          echo "About to merge 'psridharan/trigger-stage-deploy' branch into 'staging' branch"
+          # git merge --verbose main -m "Merging main branch into staging branch"
+          git merge --verbose psridharan/trigger-stage-deploy -m "Merging psridharan/trigger-stage-deploy branch into staging branch"
 
           git push origin staging
+      - name: Trigger repository_dispatch to build images
+        if: success()
+        uses: peter-evans/repository-dispatch@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          event-type: build-images
+      #- name: Trigger repository_dispatch to deploy images
+      #  if: success()
+      #  uses: peter-evans/repository-dispatch@v2
+      #  with:
+      #    token: ${{ secrets.GITHUB_TOKEN }}
+      #    event-type: deploy-images
```

Comment:
I believe a successful build image workflow will trigger the deployment on its own without a repository dispatch; it already gets around the GHA protection by using a third party GHA for triggering deployment workflows (avakar/create-deployment@v1)

Reply:
yup! it won't trigger push tests tho, so that would need a separate dispatch (though it looks like you and Andrew are discussing whether its necessary to do so)